File: Recursion/Permutation.py
def permutation(arr):
	if len(arr) <= 1:
		return arr

	perms = []

	for i in range(len(arr)):
		ele = arr[i]

		rem = arr[:i] + arr[i+1:]

		for p in permutation(rem):
			perms.append(ele+p)

	return perms

print(permutation('123'))

File: Recursion/Permutation.js
function permutation(arr){
	if(arr.length <= 1){
		return arr;
	}

	var perms = [];

	for(var i=0; i<arr.length; i++){
		var char = arr[i];
		
		var rem = arr.slice(0,i) + arr.slice(i+1, arr.length);

		for(var p of permutation(rem)){
			perms.push(char+p);
		}
		
	}

	return perms;
}

console.log(permutation('123'));